### ISSUE TITLE
Session affinity not working with HTTPProxy

### ISSUE THREAD

**Issue by movikbence** [projectcontour/contour#2004]
I'm trying to achieve Session Affinity, or sticky-session, but it seems not to be working with httpproxy.
So far I've used that strategy with IngressRoute, it was working fine.
I am using a backend application which name is kahttp, it's an open-source project as well.

Manifests:
apiVersion: projectcontour.io/v1
kind: HTTPProxy
metadata:
  name: kahttp-service
spec:
  virtualhost:
    fqdn: example.lb.com
  routes:
    - services:
        - name: kahttp-service
          port: 80
      loadBalancerStrategy:
        strategy: Cookie

I also tried that below, because it was not clear from the documentation (Migrate from ingressroute to httpproxy, and Httpproxy doc)
apiVersion: projectcontour.io/v1
kind: HTTPProxy
metadata:
  name: kahttp-service
spec:
  virtualhost:
    fqdn: example.lb.com
  routes:
  - services:
    - name: kahttp-service
      port: 80
      strategy: Cookie

So far with IngressRoute the output was:
{
  "Started": "2019-12-07T13:39:43.253925732Z",
  "Duration": 9917669842,
  "Rate": 10,
  "Clients": 1,
  "Dials": 1,
  "FailedConnections": 0,
  "Sent": 100,
  "Received": 100,
  "Dropped": 0,
  "FailedConnects": 0,
  "Hosts": {
    "kahttp-deployment-7d84b55d4d-l87sr": 100
  }

And now as you can see, all of the backend pods responded:
{
  "Started": "2019-12-07T13:41:51.613962128Z",
  "Duration": 9975871210,
  "Rate": 10,
  "Clients": 1,
  "Dials": 2,
  "FailedConnections": 0,
  "Sent": 100,
  "Received": 100,
  "Dropped": 0,
  "FailedConnects": 0,
  "Hosts": {
    "kahttp-deployment-7d84b55d4d-ftj2z": 25,
    "kahttp-deployment-7d84b55d4d-l87sr": 25,
    "kahttp-deployment-7d84b55d4d-lgjdf": 25,
    "kahttp-deployment-7d84b55d4d-rd4tj": 25
  }

- Contour version: 1.0 
- Kubernetes version: (use `kubectl version`): v1.15.2
- Kubernetes installer & version: v1.15.2
- Cloud provider or hardware configuration: Bare metal
- OS (e.g. from `/etc/os-release`): Centos

**Comment by davecheney** [comment]
@movikbence thank you for raising this issue, I'm sorry you're having difficulties. Could you please edit the issue description and use the ``` syntax to enclose your yaml samples as whitespace is important in this case.

**Comment by davecheney** [comment]
@movikbence also, in httpproxy the name of the key is
```
spec:
  routes:
  - loadBalancerPolicy:
       strategy: Cookie
```

**Comment by davecheney** [comment]
Also, I'm sorry that the documentation is completely wrong in this case. It looks like the documentation was not updated when we moved the load balancing policy to the route level.

**Comment by movikbence** [comment]
Hi

Thank you for the help, it worked with your example, you can close this issue.

**Comment by davecheney** [comment]
Reopening and assigning to myself for the docs fixes.

**Comment by davecheney** [comment]
Fixed in 96fb301
